images/image.go

```diff
@@ -210,6 +210,9 @@ func Manifest(ctx context.Context, provider content.Provider, image ocispec.Desc
 				}
 			}
 
+			// mimic behavior of client.Pull which limits to 1 manifest list child
+			descs = limitChildren(descs, platform, 1)
```

Comment:
My prior concern with having this limit here is that it requires the manifest list have set the platforms on all the manifests. In the current code it can figure out the platform from the config as well. That being said, I realize this `Manifest` function is already kind of weird and we really should update Pull to always pull manifests and configs for all platforms and just refrain from pulling layers for non-matching platforms. Is that related to needing this here?